### DISCUSSION TITLE
Binding x-transition broken after upgrading to v3

### DISCUSSION THREAD

**Discussion by ricki-io** [alpinejs/alpine#3040] [7. Bugs (archived)]
My application has a sidebar which opens from left or right depending of the user settings. The open transition has a negative/positive Tailwind animation based on the mentioned preference.

This is the code working in v2.
```
<div :x-transition:enter-start="sidebarSide === 'left' ? '-translate-x-full' : 'translate-x-full'">
```

After upgrading to v3 is not working anymore. I can't seem to find a way to fix this issue.

**Comment by Tim-Wils** [comment]
It's broken for a while now, I finally managed to get some time to look into it and I think I found a fix.
Please let me know if this fixes it:
https://codepen.io/onstuimig/pen/XWENyWY

Also fixed this one:
#1633

**Reply by ricki-io** [reply]
It works on the first print but then it's not updated.

`x-bind="sidebarSide === 'left' ? leftTransition : rightTransition"`

With the initial print, the elements gets the correct left or right transition, but when the `sidebarSide` is updated, the transition is not updated anymore. Was working with Alpine V2, broken when updating to V3.

**Reply by Tim-Wils** [reply]
@ricard-pons Are you sure? It works for me if I add your codesnippet to my codepen.
Click on Toggle, then switch directions, then toggle off en toggle again.

```
<div x-data="{ open: false, direction: 'left', leftTrans: { 'x-transition:enter-start': 'transform -translate-x-full opacity-0' }, rightTrans: { 'x-transition:enter-start':'transform translate-x-full opacity-0' } }" class="p-8 space-y-8" >
        Direction var: <button class="bg-black p-4 rounded-md px-8 py-2 text-white" @click.prevent="direction = (direction == 'left') ? 'right' : 'left'" x-text="direction">direction</button><br>
        Show: <button class="bg-black p-4 rounded-md px-8 py-2 text-white" @click.prevent="open = !open" x-text="'Toggle ' + open">Toggle </button>

        <div 
            class="relative"
            x-show="open"

            x-transition:enter="transition ease-out duration-1000"
            x-transition:enter-end="transform translate-x-0 opacity-100"

            x-bind="direction === 'left' ? leftTrans : rightTrans"

           
        >
            Content...<br>
            Content...<br>
            Content...<br>
            Content...<br>
            Content...<br>
        </div>
    </div>
```

**Reply by ricki-io** [reply]
If you replace the JS from your Codepen for a CDN with the latest version of Alpinejs, it doesn't work.

`<script defer src="https://unpkg.com/alpinejs@3.x.x/dist/cdn.min.js"></script>`

**Reply by SimoTod** [reply]
> If you replace the JS from your Codepen for a CDN with the latest version of Alpinejs, it doesn't work.
> 
> `<script defer src="https://unpkg.com/alpinejs@3.x.x/dist/cdn.min.js"></script>`

@ricard-pons I believe the whole point is that you use the js version in the codepen which has the fix to test it before @Tim-Wils raises a PR.

**Reply by ricki-io** [reply]
Sorry, I completely miss understood @Tim-Wils comment. 

I have now used the JS from the Codepen in my project and the transition is working perfectly!

**Reply by Tim-Wils** [reply]
@ricard-pons thanks for testing and confirming that it works! I will raise a PR asap

**Reply by Tim-Wils** [reply]
#3051 PR with the fix
